src/validator.spec.ts

```diff
@@ -76,7 +75,92 @@ describe('fonk-bic-validator specs', () => {
     expect(result).toEqual({
       succeeded: false,
       message: 'other message',
-      type: '',
+      type: VALIDATOR_TYPE,
+    });
+  });
+  it('should succeeded  when it has 11 length value', () => {
+    // Arrange
+    const value = 'BBVAESMM000';
+
+    setErrorMessage('');
+
+    // Act
+    const result = validator({ value });
+
+    // Assert
+    expect(result).toEqual({
+      succeeded: true,
+      message: '',
+      type: VALIDATOR_TYPE,
+    });
+  });
+  it('should succeeded  when it has 8 length value', () => {
+    // Arrange
+    const value = 'BBVAESMM';
+
+    setErrorMessage('');
+
+    // Act
+    const result = validator({ value });
+
+    // Assert
+    expect(result).toEqual({
+      succeeded: true,
+      message: '',
+      type: VALIDATOR_TYPE,
+    });
+  });
+  it('should failed  when it has incorrect length value', () => {
+    // Arrange
+    const value = 'BBVAESM';
+    const value1 = 'BBVAESM0000000';
+
+    setErrorMessage('Incorrect length');
+
+    // Act
+    const result = validator({ value });
+    const result1 = validator({ value: value1 });
+
+    // Assert
+    expect(result).toEqual({
+      succeeded: false,
+      message: 'Incorrect length',
+      type: VALIDATOR_TYPE,
+    });
+    expect(result1).toEqual({
+      succeeded: false,
+      message: 'Incorrect length',
+      type: VALIDATOR_TYPE,
+    });
+  });
+  it('should failed  when it has incorrect  value', () => {
+    // Arrange
+    const value = 'BBVA5ESM';
+    const value1 = 'BBVAESM0000000';
+    const value2 = '12345678';
+
+    setErrorMessage('Incorrect length, only can contains letters');
+
+    // Act
+    const result = validator({ value });
+    const result1 = validator({ value: value1 });
+    const result2 = validator({ value: value2 });
+
+    // Assert
+    expect(result).toEqual({
+      succeeded: false,
+      message: 'Incorrect length, only can contains letters',
+      type: VALIDATOR_TYPE,
+    });
+    expect(result1).toEqual({
+      succeeded: false,
+      message: 'Incorrect length, only can contains letters',
+      type: VALIDATOR_TYPE,
+    });
+    expect(result2).toEqual({
+      succeeded: false,
+      message: 'Incorrect length, only can contains letters',
+      type: VALIDATOR_TYPE,
     });
   });
 });
```

Comment:
Add more cases, include as many countries and cases as possible.

Valid BIC codes:

Extracted from this page (https://es.wikipedia.org/wiki/ISO_9362)

BCRDDOSX
BCVEVECAXXX
BBOGCOBBXXX
BCHICLRMXXX
BECHCLRM
BKCHHKHH
BCPLPEPL
NACNARBA 
NACNARBAXXX
BSCHESMM
BSCHESMMXXX
UNIOVECAXXX
CAIXESBB
CAIXESBBXXX
CHASUS33XXX
MANTUS33

Covering Kosovo special case (https://www.theswiftcodes.com/kosovo/), add a describe for this:
EKOMXKPR
NCBAXKPR
BPBXXKPR
CBRKXKPR
NLPRXKP
TEBKXKPR020
TEBKXKPR033

Reply:
More cases:

Add cases with incorrect country codes.

Add cases with bad formatted bic codes

Reply:
Ok, I will, I'm pending to complete the correct regular expression, and then I will do as many test cases as I can!